### DISCUSSION TITLE
wire:click issue on new DOM

### DISCUSSION THREAD

**Discussion by gzai** [livewire/livewire#4845] [1. Help] [answered]
Hello,

I'm having a problem with livewire wire:click and tippsy plugin. When click on tippsy pop up, wire:click not working.
Any suggest, to run wire:click on new DOM?

blade :
```
<div class="dropdown">
    <button class="btn btn_outlined btn_secondary" data-toggle="dropdown-menu">
        {{ $datatables_show_items }}
        <span class="ltr:ml-3 rtl:mr-3 la la-caret-down text-xl leading-none"></span>
    </button>
    <div class="dropdown-menu">
    @foreach( $datatables_list_show_items as $key => $value )
        <a href="javascript:void(0);" wire:click="changeShowItems({{ $value }})">{{ $value }}</a>
    @endforeach
     </div>
</div>
```

livewire :
```
public function changeShowItems($value)
{
    dd($value);
}
```

preview :
![wire-click-dom](https://user-images.githubusercontent.com/4225733/160758050-4c9ab666-082b-4797-a7fb-68db11f3965c.gif)

**Comment by joshhanley** [comment, marked as answer]
@gzai as the dropdown is generated using a javascript library, `wire:click` won't be applied to the correct elements, nor will it be initialised (as it is created after Livewire is running).

You will need to make use of the libraries javascript to listen for the event yourself and then use Livewire's javascript API to send the updated data.

Checkout https://laravel-livewire.com/docs/2.x/alpine-js#interacting-with-livewire-from-alpine and https://laravel-livewire.com/docs/2.x/inline-scripts#accessing-javascript-component-instance

Hope this helps!

**Reply by gzai** [reply]
Yes it is quite helpful @joshhanley. Easy way using jquery instead of javascript, I don't know how to use simple javascript.
Thanks.

blade : 
```
<div class="dropdown">
    <button class="btn btn_outlined btn_secondary" data-toggle="dropdown-menu">
        {{ $datatables_show_items }}
        <span class="ltr:ml-3 rtl:mr-3 la la-caret-down text-xl leading-none"></span>
    </button>
    <div class="dropdown-menu">
    @foreach( $datatables_list_show_items as $key => $value )
        <a href="javascript:void(0);" class="datatables_show_items" data-value="{{ $value }}">{{ $value }}</a>
    @endforeach
     </div>
</div>

@push('scripts')
<script src="https://code.jquery.com/jquery-3.6.0.min.js"></script>

<script>
    $(function() {
        $('body').on('click', '.datatables_show_items', function() {
            var $this = $(this);
            var $value = $this.data('value');
            window.livewire.emit('changeShowItems', $value);
        });
    });
</script>
@endpush
```

livewire : 
```
protected $listeners = ['changeShowItems'];

public function changeShowItems($item)
{
    dd( $item );
}
```
